src/routes/docs/tutorials/apple/step-5/+page.markdoc

```diff
@@ -0,0 +1,402 @@
+        }
+        
+    }
+}
+
+
+extension Route: Hashable {
+    static func == (lhs: Route, rhs: Route) -> Bool {
+        return lhs.compareString == rhs.compareString
+    }
+    
+    var compareString: String {
+        switch self {
+        case .appwrite:
+            return "appwrite"
+        case .home:
+            return "home"
+        case .auth:
+            return "auth"
+        case .add_habit:
+            return "addHabit"
+        }
+    }
+}
+
+
+final class Router: ObservableObject {
+    @Published var routes = [Route]()
+    @Published var selectedTab: Tabs = .home
+    
+    func push(_ screen: Route) {
+        routes.append(screen)
+    }
+    
+    func pushReplacement(_ screen: Route) {
+        if routes.isEmpty {
+            routes.append(screen)
+        } else {
+            routes[routes.count - 1] = screen
+        }
+    }
+    
+    func pop() {
+        routes.removeLast()
+    }
+    
+    func popUntil(predicate: (Route) -> Bool) {
+        if let last = routes.popLast() {
+            guard predicate(last) else {
+                popUntil(predicate: predicate)
+                return
+            }
+        }
+    }
+    
+    func reset() {
+        routes = []
+    }
+}
+```
+
+# ContentView {% #contentview %}
+
+Update the ContentView.swift to reflect some changes related to our navigation
+
+```swift
+import SwiftUI
+
+struct ContentView: View {
+    
+    @EnvironmentObject private var router: Router
+    @EnvironmentObject private var userViewModel: UserViewModel
+    @EnvironmentObject private var snackBarService: SnackBarService
+    @Environment(\.colorScheme) private var theme
+    
+    
+    var body: some View {
+        NavigationStack(path: $router.routes) {
+            VStack {
+                
+                Image(theme == .dark ? "appwrite_light" : "appwrite_dark")
+                    .frame(width: 132, height: 24)
+                
+            }
+            .task {
+                
+                let isAuthenticated = await userViewModel.getCurrentSession()
+                
+                if !isAuthenticated {
+                    router.pushReplacement(.auth)
+                } else {
+                    router.pushReplacement(.home)
+                }
+            }
+            .navigationDestination(for: Route.self, destination: { $0 })
+            
+        }
+        .overlay(alignment: .top) {
+            if (snackBarService.snackBarState?.hasError == true) {
+                SnackbarView(text: snackBarService.snackBarState?.error.localizedDescription ?? "An error occured")
+            }
+        }
+        
+    }
+}
+```
+
+Note: You can get the Appwrite logo on Appwrite website and add it to your project assets
+
+# Inject services and view models {% #inject-services-and-view-models %}
+
+Let's inject Appwrite Service and Snarbar Service into our View Model then inject to our app using Environment Object. Update the entry file with the code below.
```

Comment:
Tbh, instead of entry file, why don't you suggest a name we use when following this project from beginning to end. It'll be easier than doing this + saying app_hacktoberfest